h2o-core/src/main/java/water/parser/ParseSetup.java

```diff
@@ -622,25 +623,17 @@ private static final void checkEncoding(byte[] bits) {
    * @param bytes Array of bytes (containing 0 or more newlines)
    * @return The longest line length in the given bytes
    */
-  private static final long maxLineLength(byte[] bytes) {
-    if (bytes.length >= 2) {
-      String st = new String(bytes);
-      StringReader sr = new StringReader(st);
-      BufferedReader br = new BufferedReader(sr);
-      String line;
-      long maxLineLength=0;
-      try {
-        while(true) {
-          line = br.readLine();
-          if (line == null) break;
-          maxLineLength = Math.max(line.length(), maxLineLength);
-        }
-      } catch (IOException e) {
-        return -1;
+  private static final int maxLineLength(byte[] bytes) {
```

Comment:
You mean Vlad also made similar change? Max line length is only used in the chunk size heuristic, so I guess it does not have to be super precise. But the old method was really slow, allocating new String for each line. It would not matter for single file, but it can be a considerable burden for multifile case. Also, for some reason it was way off on AirlinesTrain from small data. It came back with super long line filled with garbage which is clearly not in the data. 

I believe this code belongs to the parser itself. It should be left to the parser to figure out where he thinks is the longest line. We should probably add a field into ParseSetup and let each parser to figure this out. I did not want to change more than necessary to fix this issue though.